cmd/snap-update-ns/mount-entry-change.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "mount-entry-change.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "../libsnap-confine-private/utils.h"
+
+const char *sc_mount_action_to_str(enum sc_mount_action action)
+{
+	switch (action) {
+	case SC_ACTION_NONE:
+		return "none";
+	case SC_ACTION_MOUNT:
+		return "mount";
+	case SC_ACTION_UNMOUNT:
+		return "unmount";
+	default:
+		return "???";
+	}
+}
+
+/**
+ * Look through the haystack and find the first needle.
+ **/
+static struct sc_mount_entry *sc_mount_entry_find(struct sc_mount_entry
+						  *haystack, const struct sc_mount_entry
+						  *needle)
+{
+	for (; haystack != NULL; haystack = haystack->next) {
+		if (sc_compare_mount_entry(needle, haystack) == 0) {
+			return haystack;
+		}
+	}
+	return NULL;
+}
+
+static struct sc_mount_change *sc_mount_change_alloc()
+{
+	struct sc_mount_change *change = calloc(1, sizeof *change);
+	if (change == NULL) {
+		die("cannot allocate sc_mount_change object");
+	}
+	return change;
+}
+
+static void sc_mount_change_free_chain(struct sc_mount_change *change)
+{
+	struct sc_mount_change *c = change, *n;
+	while (c != NULL) {
+		n = c->next;
+		free(c);
+		c = n;
+	}
+}
+
+struct sc_mount_change *sc_compute_required_mount_changes(struct sc_mount_entry_list
+							  *desired, struct
+							  sc_mount_entry_list
+							  *current)
```

Comment:
This is pretty hard to read. The full type and the variable name should be on the same line instead of blindly wrapping to respect a pretty short terminal width (which most of us don't use anymore):
```
struct sc_mount_change *sc_compute_required_mount_changes(
	struct sc_mount_entry_list *desired,
	struct sc_mount_entry_list *current)
```
Same for the function above.